src/main/java/TaskSchedular.java

```diff
@@ -0,0 +1,7 @@
+
+public interface TaskSchedular {
+    void addNewTaskToQueueAndMap(Task task);
+    void printTaskQueue();
+    Task fetchTaskFromQueueForExecution();
+    void printTaskGroupsBasedOnPriority();
+}
```

Comment:
Some of the method names breaks SR principle 
It seems implementation for these methods are similar in some way. Is these any chances for refactoring?